### DISCUSSION TITLE
Error on `non-nullable` fields

### DISCUSSION THREAD

**Discussion by saihaj** [graphql/defer-stream-wg#22] [Q&A] [answered]
Today with `graphql` if there is a `Non-Nullable` field that returns `null` executor will throw an error `Cannot return null for non-nullable field FIELDNAME`. But if I try to `@defer` a `non-nullable` field which returns `null` I don't get any error back on client back. Shouldn't it return an error back? 

Here is a code sandbox: https://codesandbox.io/s/defer-stream-errors-lisfo?file=/src/index.mjs

**Comment by robrichard** [comment, marked as answer]
@saihaj yes you should get these errors. There were some bugs in the handling of non-null errors that should be fixed now, and I also added some tests for these cases. Let me know if you still run into any issues with the latest code on the `defer-stream` branch.

The way non-nullable errors bubble will need to be treated differently - opened a separate topic for that: #23
